### PR TITLE
Updated installation instructions

### PR DESCRIPTION
**Pull request by aulihant**
Followed steps for offline install and developer's installation to make sure it works with Ubuntu 20.04

### CODE REVIEW

File: docs/user_guide/installation.rst [hunk 1]
```diff
@@ -68,29 +74,77 @@ From the command shell, enter the following sequence to create and activate a vi
...
+--------------------------
+
+.. _SC Install:
+
+
+After you've got the python dependencies installed, you will need to install SiliconCompiler. There are a few different ways to do this:
+
+1. The recommended method is to install the last stable version published to `pypi.org <https://pypi.org>`_, or
+2. You can do an offline install with a tarball (for Linux only), or
+3. You can install directly from the git repository (best for developers).
```
Comment (gadfort) [inline docs/user_guide/installation.rst:89]: Add links to each section describing these

File: docs/user_guide/installation.rst [hunk 2]
```diff
@@ -68,29 +74,77 @@ From the command shell, enter the following sequence to create and activate a vi
...
+
+Install from pypi.org 
+^^^^^^^^^^^^^^^^^^^^^
+SiliconCompiler can be installed directly from `pypi.org <https://pypi.org>`_ using pip. Activate your `Python Virtual Environment <https://docs.python.org/3/library/venv.html>`_ and follow the instructions below. (identical for Windows, Linux, and macOS).
```
Comment (gadfort) [inline docs/user_guide/installation.rst:93]: no need for the "(identical for Windows, Linux, and macOS)." statement

File: docs/user_guide/installation.rst [hunk 3]
```diff
@@ -68,29 +74,77 @@ From the command shell, enter the following sequence to create and activate a vi
...
 
 .. code-block:: bash
 
  (venv) pip install --upgrade pip                # upgrade pip in virtual env
  (venv) pip list                                 # show installed packages in venv
  (venv) pip install --upgrade siliconcompiler    # install SiliconCompiler in venv
  (venv) python -m pip show siliconcompiler       # will display  SiliconCompiler package information
- (venv) python -c "import siliconcompiler;print(siliconcompiler.__version__)"
```
Comment (gadfort) [inline docs/user_guide/installation.rst:82]: do we want to show an example output from this command?

File: docs/user_guide/installation.rst [hunk 4]
```diff
@@ -68,29 +74,77 @@ From the command shell, enter the following sequence to create and activate a vi
...
 
-   \ |release|
+To access them:
+
+1. Go our  `builds page <https://github.com/siliconcompiler/siliconcompiler/actions/workflows/wheels.yml>`_. 
+2. Click on the most recent, passing Wheels package. This should be the first green-colored build in the list.
+3. On the bottom of that page, you will see an "Artifacts" section. Click on the "artifact" to download it.
+4. The packages are named ``scdeps-<pyversion>.tar.gz``, depending on which Python version they are associated with.
+
+To install from a bundle, create a Python virtual environment following the instructions above, then perform the following commands.
```
Comment (gadfort) [inline docs/user_guide/installation.rst:115]: link back to that section

File: docs/user_guide/installation.rst [hunk 5]
```diff
@@ -68,29 +74,77 @@ From the command shell, enter the following sequence to create and activate a vi
...
+.. code-block:: bash
+
+   tar -xzvf scdeps-<pyversion>.tar.gz
+   pip install --upgrade pip --no-index --find-links scdeps
+   pip install siliconcompiler --no-index --find-links scdeps
+
+.. note::
+
+Note 
+This only installs the SC build system offline. Before you can start running SiliconCompiler, you will also need to make sure you have installed external PDKs and tools required to build (synthesis, place and route, etc). Typically, users of this flow have already set up their own tools and PDKs. If that is the case, you may skip ahead to  `confirm your installation`_. But if you do not have PDKs and tools set up, have an internet connection, and would like to leverage the PDKs and external tool install information from the SiliconCompiler GitHub repo, you can, by following these steps:
```
Comment (gadfort) [inline docs/user_guide/installation.rst:126]: I don't think the PDKs are in the wheel builds, so this applies to pip installs too.

File: docs/user_guide/installation.rst [hunk 6]
```diff
@@ -103,30 +157,40 @@ Ensure that the path to the Homebrew packages takes priority over system
...
+.. parsed-literal::
+
+   \ |release|
+
+
+Running SiliconCompiler
+------------------------------
+
+Run Remote with Cloud Access
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (gadfort) [inline docs/user_guide/installation.rst:193]: On line 205, make sc-configure a :ref:`sc-configure` and same for sc-ping

File: docs/user_guide/installation.rst [hunk 7]
```diff
@@ -164,6 +228,24 @@ Once you have verified that your remote configuration works, try compiling a sim
    (venv) curl https://raw.githubusercontent.com/siliconcompiler/siliconcompiler/main/docs/user_guide/examples/heartbeat.v > heartbeat.v
    (venv) sc heartbeat.v -remote
 
+
+Run Locally
+^^^^^^^^^^^
+
+.. _Run local:
+
+If you wish to run locally, you will need to install some external tool dependencies to start. Take a look at `External Tools`_ to start with Surelog, Yosys, OpenRoad and KLayout as a minimum to run.
```
Comment (gadfort) [inline docs/user_guide/installation.rst:237]: I think this should be clarified to say that those tools are required for as a minimum for a ASIC flow

File: docs/user_guide/installation.rst [hunk 8]
```diff
@@ -177,6 +259,7 @@ To test the klayout installation, run the 'sc-show' to display the 'heartbeat' l
 
 External Tools
 --------------
+.. _External Tools:
```
Comment (gadfort) [inline docs/user_guide/installation.rst:262]: Change last line from ending with "here" to make the "tools directory" the link

File: docs/user_guide/installation.rst [hunk 9]
```diff
@@ -164,6 +228,24 @@ Once you have verified that your remote configuration works, try compiling a sim
...
+If you wish to run locally, you will need to install some external tool dependencies to start. Take a look at `External Tools`_ to start with Surelog, Yosys, OpenRoad and KLayout as a minimum to run.
+
+Once you have these tools installed, try compiling a simple design:
+
+.. code-block:: bash
+
+    (venv) cd $SCPATH/../examples/heartbeat
+    (venv) sc heartbeat.v heartbeat.sdc
+
+See the _`Quickstart <https://docs.siliconcompiler.com/en/latest/user_guide/quickstart.html>`_ section to get more details on what you're running.
```
Comment (gadfort) [inline docs/user_guide/installation.rst:246]: This should just be a :ref:quickstart

File: docs/user_guide/installation.rst (hunk 6 above)
Comment (gadfort) [inline docs/user_guide/installation.rst:193]: Add a link to the remote processing section for more information

**Review by gadfort — CHANGES_REQUESTED** [review]
There are two error when building:
siliconcompiler/docs/user_guide/installation.rst:123: ERROR: Content block expected for the "note" directive; none found.
siliconcompiler/docs/user_guide/installation.rst:246: WARNING: Inline target start-string without end-string.

File: docs/user_guide/installation.rst [hunk 10]
```diff
@@ -103,32 +165,40 @@ Ensure that the path to the Homebrew packages takes priority over system
...
    python -m pip install -e .
+   export SCPATH=<the full path for your siliconcompiler/siliconcompiler directory>
 
-Offline Install (Linux only)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-We also provide packages that bundle SC with all of its Python dependencies to enable installation on machines without an external internet connection. They can be found under the "Artifacts" section of any passing nightly or release build on our `builds page <https://github.com/siliconcompiler/siliconcompiler/actions/workflows/wheels.yml>`_. The packages are named ``scdeps-<pyversion>.tar.gz``, depending on which Python version they are associated with.
 
-To install from a bundle, create a Python virtual environment following the instructions above, then perform the following commands.
+Confirm your Installation
```
Comment (aulihant) [inline docs/user_guide/installation.rst:181]: @gadfort this is an example of the breaking out a section that is common to all methods of installation. This prevents copy/pasting of instruction thus reducing the likelihood of divergence. But it also potentially makes instructions more confusing for someone who wants to just read all the way through without skipping around. Do you have any thoughts on this?

**Comment by aulihant** [comment]
@gadfort I incorporated your requested changes which are really straightforward. The two comments which are left unresolved are related to general documentation flow. Some instructions will be the same for different install methods (e.g. checking your installation, needing PDKs and external tools installed), so we can either create common instruction for these, or copy and paste the same instruction into different sections.

File: docs/user_guide/installation_confirm_version.rst [hunk 11]
```diff
@@ -0,0 +1,13 @@
+To confirm your installation:
```
Comment (gadfort) [inline docs/user_guide/installation_confirm_version.rst:1]: Can you move these into a subdirectory like docs/user_guide/installation/?

File: docs/user_guide/installation_prep_path.rst [hunk 12]
```diff
@@ -0,0 +1,6 @@
+To prepare your path and get set up with examples and PDKs to run:
```
Comment (gadfort) [inline docs/user_guide/installation_prep_path.rst:1]: Can you move these into a subdirectory like docs/user_guide/installation/?

File: docs/user_guide/installation.rst [hunk 13]
```diff
@@ -68,29 +76,87 @@ From the command shell, enter the following sequence to create and activate a vi
...
+   (venv) tar -xzvf scdeps-<pyversion>.tar.gz
+   (venv) pip install --upgrade pip --no-index --find-links scdeps
+   (venv) pip install siliconcompiler --no-index --find-links scdeps
+
+.. include:: installation_confirm_version.rst
+
+.. note::
+
+   Before you can start running SiliconCompiler, you will also need to make sure you have installed external PDKs and tools required to build (synthesis, place and route, etc). Typically, users of this flow have already set up their own tools and PDKs.
+   If you need to set up PDKs and tools and have an internet connection, you can run the following steps.
```
Comment (gadfort) [inline docs/user_guide/installation.rst:139]: I think you should add this to the prep_path as well.

File: docs/user_guide/installation.rst [hunk 14]
```diff
@@ -68,29 +76,87 @@ From the command shell, enter the following sequence to create and activate a vi
...
+
+   .. include:: installation_prep_path.rst
 
-.. parsed-literal::
 
-   \ |release|
+Skip to `Run local`_.
```
Comment (gadfort) [inline docs/user_guide/installation.rst:144]: Should this be a link to Run instead of Run local?

File: docs/user_guide/installation.rst [hunk 15]
```diff
@@ -103,32 +169,32 @@ Ensure that the path to the Homebrew packages takes priority over system
...
+------------------------------
 
-   tar -xzvf scdeps-<pyversion>.tar.gz
-   pip install --upgrade pip --no-index --find-links scdeps
-   pip install siliconcompiler --no-index --find-links scdeps
+You can either run remotely in the cloud, or run locally on your machine.
 
-Cloud Access
---------------
+Run Remote with Cloud Access
```
Comment (gadfort) [inline docs/user_guide/installation.rst:194]: Lets discuss this one tomorrow

File: docs/user_guide/installation.rst [hunk 16]
```diff
@@ -164,22 +230,46 @@ Once you have verified that your remote configuration works, try compiling a sim
...
+    (venv) sc heartbeat.v heartbeat.sdc
+
+See the :ref:`Quickstart guide <quickstart guide>` section to get more details on what you're running.
+
+
 Layout Viewer
 -------------
 
-To view IC layout files (DEF, GDSII) we recommend installing the open source multi-platform 'klayout' viewer (available for Windows, Linux, and macOS). Installation instructions for klayout can be found `HERE <https://www.klayout.de/build.html>`_.
+To view IC layout files (DEF, GDSII) we recommend installing the open source multi-platform 'klayout' viewer (available for Windows, Linux, and macOS). Installation instructions for klayout can be found in the :ref:`tools directory <klayout>`.
```
Comment (gadfort) [inline docs/user_guide/installation.rst:255]: You can just make this a link to klayout with :ref:`klayout`

File: docs/user_guide/installation.rst [hunk 17]
```diff
@@ -164,22 +230,46 @@ Once you have verified that your remote configuration works, try compiling a sim
...
 
 To test the klayout installation, run the 'sc-show' to display the 'heartbeat' layout:
```
Comment (gadfort) [inline docs/user_guide/installation.rst:257]: Ref to sc-show

**Review by gadfort — CHANGES_REQUESTED** [review]
Looks good, we can chat about the last couple of things tomorrow morning

**Review by gadfort — APPROVED** [review]

**Pull request merged**